### PR TITLE
feat(add): add Patreon button

### PR DESCRIPTION
**Pull request by ahmaxed**
This pr adds a Patreon button to the rest of our donation CTA's.

After clicking on the button it should open a new tab and redirect to patreon auth (if users are not logged in) or redirect to a 5 dollar patreon pledge confirmation page.

Then, if the donation is successful, it should redirect to our donation success page.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/Donation/patreon-button.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const redirectUri = `&redirect_uri=${donationUrls.successUrl}`;
+  const href = `https://www.patreon.com/oauth2/become-patron?response_type=code${pledgeLevel}${clientId}${redirectUri}${v2Params}`;
+
+  return (
+    <a
+      className='patreon-button link-button'
+      data-patreon-widget-type='become-patron-button'
+      href={href}
+      target='_blank'
+    >
```
Comment (nhcarrigan) [inline client/src/components/Donation/patreon-button.tsx:34]: The bot is complaining about the lack of a `noreferrer` here.

Comment (ahmadabdolsaheb) [reply client/src/components/Donation/patreon-button.tsx]: Done.

**Review by naomi-lgbt — DISMISSED** [review]
This looks good to me!

File: client/src/components/Donation/DonateForm.tsx [hunk 2]
```diff
@@ -334,8 +335,9 @@ class DonateForm extends Component<DonateFormProps, DonateFormComponentState> {
             isPaypalLoading={loading.paypal}
             isSignedIn={isSignedIn}
             onDonationStateChange={this.onDonationStateChange}
-            theme={defaultTheme ? defaultTheme : theme}
+            theme={priorityTheme}
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/DonateForm.tsx:338]: I don't think the defaultTheme should take precedence over the theme, is that the intention?  Normally the default should be the fallback that gets overridden.

File: config/read-env.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ const {
   ALGOLIA_API_KEY: algoliaAPIKey,
   STRIPE_PUBLIC_KEY: stripePublicKey,
   PAYPAL_CLIENT_ID: paypalClientId,
+  PATREON_CLINET_ID: patreonClientId,
```
Comment (ojeytonwilliams) [inline config/read-env.js:32]:
```suggestion
  PATREON_CLIENT_ID: patreonClientId,
```

File: client/src/components/Donation/patreon-button.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  if (
+    !patreonClientId ||
+    !patreonDefaultPledgeAmount ||
+    !donationUrls.successUrl
+  ) {
+    return null;
+  }
+
+  const clientId = `&client_id=${patreonClientId}`;
+  const pledgeLevel = `$&min_cents=${String(patreonDefaultPledgeAmount)}`;
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/patreon-button.tsx:23]:
```suggestion
  const pledgeLevel = `$&min_cents=${patreonDefaultPledgeAmount}`;
```
TS doesn't seem to care.

File: sample.env [hunk 5]
```diff
@@ -42,6 +42,9 @@ PAYPAL_VERIFY_WEBHOOK_URL=https://api.sandbox.paypal.com/v1/notifications/verify
 PAYPAL_API_TOKEN_URL=https://api.sandbox.paypal.com/v1/oauth2/token
 PAYPAL_WEBHOOK_ID=webhook_id_from_paypal_dashboard
 
+#Patreon
+PATREON_CLINET_ID=id_from_patreon_dashboard
```
Comment (ojeytonwilliams) [inline sample.env:46]:
```suggestion
# Patreon
PATREON_CLIENT_ID=id_from_patreon_dashboard
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nitpicks aside, this looks good :)

File: client/src/components/Donation/DonateForm.tsx (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/src/components/Donation/DonateForm.tsx:338]: So on components such as the certification, we pass the defaulthTeme props down to the DonateForm. As a result we show the default theme for that page regardless of the user's theme preferences.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Wrangling about the meaning of 'default' shouldn't block this, so I'm happy to merge as is.

File: client/src/components/Donation/DonateForm.tsx (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/src/components/Donation/DonateForm.tsx:338]: Sure, it makes sense. TBH I didn't want to touch on themes on this pr, but it was on the line above what I was working on. :)

Comment (ojeytonwilliams) [reply client/src/components/Donation/DonateForm.tsx:338]: Yeah, understood. I just figured it was worth a quick discussion.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**